src/store/slices/app/index.ts

```diff
@@ -0,0 +1,54 @@
+import { PayloadAction, createSlice } from '@reduxjs/toolkit';
+import { initialState } from './initialState';
+
+const appSlice = createSlice({
+  name: 'app',
+  initialState,
+  reducers: {
+    addNotification: (
+      state,
+      action: PayloadAction<{
+        name: string;
+        source: string;
+        timeout: number | null;
+        url: string | null;
+      }>
+    ) => {
+      const now = Date.now();
+      const defaultTimeout = 5000;
+      state.notifications.push({
+        ...action.payload,
+        id: Math.floor(Math.random() * 1e6),
+        seen: false,
+        timeout: action.payload.timeout ?? now + defaultTimeout,
+      });
+    },
+    readNotification: (
```

Comment:
Lets use the same naming for variables and functions changing them. If someone has seen the notification, it doesnt mean that it was read